rpcs3/main.cpp

```diff
@@ -470,11 +477,32 @@ int main(int argc, char** argv)
 	}
 #endif
 
+	// If we're in headless, or non-gui, or we pass in the command line to exit after running, then we exit the
+	// application when the Emulator terminates (potentially returning success / error return code)
+	bool exit_after = parser.isSet(arg_headless) || parser.isSet(arg_no_gui) || parser.isSet(arg_exit_after);
+
+	if(parser.isSet(arg_install_pkgs) && !parser.value(arg_install_pkgs).isEmpty())
+	{
+		const auto pkg_paths = parser.value(arg_install_pkgs).split(',');
+		for (const auto pkg_path : pkg_paths)
+		{
+			if (!pkg_path.isEmpty())
+			{
+				if (!Emu.InstallPkg(pkg_path.toStdString()) && exit_after)
+				{
+					// We bail out, installation of packages failed...
+					// TODO: Logging to the command line of the errors
+					app->exit(1);
```

Comment:
There is no event loop running, so calling exit here should be useless.
A return would make more sense.
But it doesn't make sense anyway. If I call this regularly with a gui, then the gui will just close and I have no idea why

Reply:
"If I call this regularly with a gui, then the gui will just close and I have no idea why" huh??
What gui?  you will have called it with --headless or --no-gui or at least --exit-after (otherwise exit_after isn't set, which is a precondition for the app->exit()

Agreed that doing app->exit(1) and waiting for it to have tried to load a Game, and fire up the event loop isn't right however.